content/v1.12/concepts/managed-resources.md

```diff
@@ -3,630 +3,745 @@ title: Managed Resources
+Publishing secrets data to an external secret store like 
+[HashiCorp Vault](https://www.vaultproject.io/) relies on a 
+{{<hover label="configref" line="8">}}publishConnectionDetailsTo.configRef{{</hover>}}. 
 
-> Note: provisioning an Azure Database for PostgreSQL requires the presence of a
-> [Resource Group] in your Azure account. We go ahead and provision a new
-> `ResourceGroup` here in case you do not already have a suitable one in your
-> account.
+The 
+{{<hover label="configref" line="9">}}configRef.name{{</hover>}} references a 
+{{<hover label="storeconfig" line="4">}}StoreConfig{{</hover>}}
+object. 
 
-```yaml
-apiVersion: azure.crossplane.io/v1alpha3
-kind: ResourceGroup
-metadata:
-  name: sqlserverpostgresql-rg
-spec:
-  location: West US 2
----
-apiVersion: database.azure.crossplane.io/v1beta1
-kind: PostgreSQLServer
-metadata:
-  name: sqlserverpostgresql
+```yaml {label="configref"}
+apiVersion: rds.aws.upbound.io/v1beta1
+kind: Instance
 spec:
   forProvider:
-    administratorLogin: myadmin
-    resourceGroupNameRef:
-      name: sqlserverpostgresql-rg
-    location: West US 2
-    sslEnforcement: Disabled
-    version: "9.6"
-    sku:
-      tier: GeneralPurpose
-      capacity: 2
-      family: Gen5
-    storageProfile:
-      storageMB: 20480
-  writeConnectionSecretToRef:
-    namespace: crossplane-system
-    name: sqlserverpostgresql-conn
+  # Removed for brevity
+  publishConnectionDetailsTo:
+    name: rds-kubernetes-secret
+    configRef: 
+      name: my-vault-storeconfig
 ```
 
-```console
-kubectl apply -f https://raw.githubusercontent.com/crossplane/crossplane/release-1.10/docs/snippets/provision/azure.yaml
+```yaml {label="storeconfig"}
+apiVersion: secrets.crossplane.io/v1alpha1
+kind: StoreConfig
+metadata:
+  name: my-vault-storeconfig
+# Removed for brevity
 ```
 
-Creating the above instance will cause Crossplane to provision a PostgreSQL
-database instance on Azure. You can view the progress with the following
-command:
+{{<hint "tip" >}}
+Read the 
+[Vault as an External Secrets Store]({{<ref "knowledge-base/integrations/vault-as-secret-store">}})
+guide for details on using StoreConfig objects.
+{{< /hint >}}
 
-```console
-kubectl get postgresqlserver sqlserverpostgresql
-```
 
-When provisioning is complete, you should see `READY: True` in the output. You
-can take a look at its connection secret that is referenced under
-`spec.writeConnectionSecretToRef`:
+## Annotations
 
-```console
-kubectl describe secret sqlserverpostgresql-conn -n crossplane-system
-```
+Crossplane applies a standard set of Kubernetes `annotations` to managed
+resources.
 
-You can then delete the `PostgreSQLServer`:
+{{<table "table table-sm">}}
+| Annotation | Definition | 
+| --- | --- | 
+| `crossplane.io/external-name` | The name of the managed resource inside the Provider. |
+| `crossplane.io/external-create-pending` | The timestamp of the last time the managed resource was in the `PENDING` state. | 
```

Comment:
is `PENDING` a real state that all MRs go into? I didn't see it mentioned in the Conditions section below. Should we also mention "creation" in this text to reinforce that this annotation has to do with the creation of the MR?

Reply:
This sent me on a fun little rathole and it's what I get for blindly trusting code comments. There is no `PENDING` state; this timestamp is written when we first started trying to create a MR. A second annotation, either `external-create-succeeded` or `external-create-failed` is applied during reconciliation. 

The text of all three have been updated to be more clear. Specifically for pending:

>The timestamp of when Crossplane began creating a new managed resource.

Reply:
external-create-pending can be a blocking annotation that needs to be removed manually - see https://github.com/crossplane/crossplane-runtime/issues/340 and https://github.com/crossplane/crossplane/issues/3037